### DISCUSSION TITLE
Shell with CONFIG_SHELL_HELP disabled includes help command and prints root commands

### DISCUSSION THREAD

**Discussion by timwoolliscroft-opteran** [zephyrproject-rtos/zephyr#74068] [Q&A]
If the CONFIG_SHELL_HELP KConfig is disabled developer is likely trying to hide all help from the user. However the behaviour is to include the command.

By inspection:
https://github.com/zephyrproject-rtos/zephyr/blob/0d56a3c3f4c333b08b0fecde02a9c032ee6c2d2f/subsys/shell/shell.c#L1808C17-L1808C34
The command is left in, and prints the root command list

**Comment by aescolar** [comment]
CC @jakub-uC

**Comment by jakub-uC** [comment]
@timwoolliscroft-opteran 

The main purpose of the Kconfig option: `CONFIG_SHELL_HELP` is to enable or disable support for the help shell module. That is, if this option is active, the shell itself parses and looks for the `-h` or `--help` options. Additionally, functions for nicely formatting help messages for individual commands and their subcommands are compiled.

I somewhat unfortunately named the command `help`. The request for its implementation came from users who do not have the ability to send VT100 commands, and therefore cannot use the Tab button to list existing root commands. The help command is intended for such cases. If it indeed causes any issues, I propose creating a new Kconfig option that disables it.
